pkg/controller/pv_controller_sync.go

```diff
@@ -0,0 +1,219 @@
+/*
+Copyright © 2021 The MayaData Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controller
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	collectorinterface "github.com/mayadata-io/volume-events-exporter/pkg/collectorinterface"
+	"github.com/mayadata-io/volume-events-exporter/pkg/collectorinterface/tokenauth"
+	"github.com/mayadata-io/volume-events-exporter/pkg/nfspv"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	k8serror "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+)
+
+const (
+	annotationProcessEventKey    = "events.openebs.io/required"
+	eventRequiredAnnotationValue = "true"
+)
+
+// processVolumeEvents reconciles PersistentVolume and will
+// send volume information to configured callback URL if it is required
+// send (or) volume event information is not yet sent
+func (pController *PVEventController) processVolumeEvents(key string) (bool, error) {
+	klog.V(4).Infof("Started syncing PV: %s to send send metrics information", key)
+
+	// Convert the key string into a distinct namespace and name
+	_, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		runtime.HandleError(fmt.Errorf("invalid resource key: %s", key))
+		return false, nil
+	}
+
+	// Get PV resource with above name
+	pvObj, err := pController.kubeClientset.CoreV1().PersistentVolumes().Get(context.TODO(), name, metav1.GetOptions{})
+	if k8serror.IsNotFound(err) {
+		runtime.HandleError(fmt.Errorf("PV %q has been deleted", key))
+		return false, nil
+	}
+	if err != nil {
+		return false, err
+	}
+
+	err = pController.sync(pvObj)
+	// TODO: If events needs to be generated undo the comments
+	// if err != nil {
```

Comment:
lets enable the event

Reply:
If we enable the events, then users will get to know we are sending events somewhere. Is it fine?

Reply:
Enabled based on `--generate-k8s-event` flag value